src/node/snapshotter.h

```diff
@@ -248,14 +258,23 @@ namespace ccf
       std::lock_guard<std::mutex> guard(lock);
 
       CCF_ASSERT_FMT(
-        idx >= next_snapshot_indices.back(),
+        idx >= next_snapshot_indices.back().idx,
         "Committable seqno {} < next snapshot seqno {}",
         idx,
-        next_snapshot_indices.back());
+        next_snapshot_indices.back().idx);
+
+      bool force = store->flag_enabled(
+        kv::AbstractStore::Flag::SNAPSHOT_AT_NEXT_SIGNATURE);
 
-      if ((idx - next_snapshot_indices.back()) >= snapshot_tx_interval)
+      if (force)
       {
-        next_snapshot_indices.push_back(idx);
+        store->unset_flag(kv::AbstractStore::Flag::SNAPSHOT_AT_NEXT_SIGNATURE);
+        schedule_snapshot(idx);
```

Comment:
Scheduling the snapshot at this point is too early since the signature may still be rolled back. This is why we only create snapshots when the snapshotted index cannot be rolled back (i.e. on commit).

Reply:
So that would be in `record_signature` then?

Reply:
I believe what you had previously should work with some minor tweaks:
1. Primary snapshotter pushes `{idx, true}` in `next_snapshot_indices` when a snapshot is forced (`record_committable()` doesn't even need to return `true` as the ledger chunk will be forced by the entry header anyway).
2. On `commit(commit_idx)`, if `commit_idx >= next_snapshot_indices.back().idx`, then a snapshot can be scheduled because it's forced (even though the `snapshot_tx_interval` hasn't passed).
3. The backup node doesn't need to know a snapshot was created at this index, since the ledger chunk will have been automatically created because of the ledger entry header.

In other words, we decouple the force snapshot generation from the regular interval generation, which is fine.

Reply:
Isn't that exactly what I had before? The problem was that the snapshot interval on the backups would get out of sync. I forget where exactly that was, but even after fixing the `last_snapshot_idx` to not be updated for forced snapshots, it still got out of sync.

Reply:
OK, I think I got it now. It took some more precision for the cases in which snapshot interval and forced index coincide. Waiting for the CI now and if it passes this is ready to be merged.